web_app

xma4602 opened the pull request.

xma4602 committed:
Разделены модель и сущности. Под них переделаны тесты.

xma4602 committed:
Созданы sql-скрипты на создание БД, таблиц.
Сгенерированы тестовые данные в json и переведены в формат sql для вставки в БД.

xma4602 committed:
Изменены sql-скрипты добавление данных.
В surveys появился title.

xma4602 committed:
Написаны сущности БД.
В pom.xml добавлены зависимости hibernate & postgresql

xma4602 committed:
Перемещен SurveyManager

xma4602 committed:
Сделаны сущности БД

xma4602 committed:
Сделаны репозитории DAO БД

xma4602 committed:
конфигурация Hibernate

xma4602 committed:
Доработки сущностей и репозиториев

xma4602 committed:
Написаны НЕРАБОЧИЕ тесты

xma4602 committed:
Переделаны сущности и репозитории

xma4602 committed:
Переделаны sql-скрипты создания таблиц и вставки данных вместе с обработчиком

xma4602 committed:
Merge branch 'db_scripts' into data_access

# Conflicts:
#	.gitignore
#	src/main/resources/sql/insert_answers.sql
#	src/main/resources/sql/insert_questions.sql
#	src/main/resources/sql/insert_surveys.sql

xma4602 committed:
Исправлена проблема с отсутствием count в answers

xma4602 committed:
Merge branch 'db_scripts' into data_access

xma4602 committed:
В скриптах добавления тестовых данных в бд добавлены статус и тип для вопросов.

xma4602 committed:
Доделаны скрипты для тестов репозиториев

xma4602 committed:
Доделаны тесты репозиториев и в соответствии с ними исправлены ошибки в сущностях.

xma4602 committed:
К опросу добавлен заголовок

xma4602 committed:
Скрипты помещены в нужные папки

xma4602 committed:
В ветке удалено все не относящееся к web-части

xma4602 committed:
Все файлы по созданию БД вынесены в папку db_init

xma4602 committed:
Revert "Все файлы по созданию БД вынесены в папку db_init"

This reverts commit b4a0cc4b77d975c9a9c60e5b8b411bc1ef73accc.

xma4602 committed:
Все файлы БД вынесены в db_init

xma4602 committed:
Удалено все, что не относится к db_init

xma4602 committed:
страница анкеты

xma4602 committed:
страница анкеты 2

xma4602 committed:
страницы отображения таблиц опросов, вопросов и ответов

xma4602 committed:
Merge branch 'db_scripts' into data_access

# Conflicts:
#	src/db_init/hibernate.cfg.xml
#	src/db_init/sql_inserts_generator.py
#	src/main/java/com/xma/surveys/model/Question.java
#	src/main/java/com/xma/surveys/model/QuestionStatus.java
#	src/main/java/com/xma/surveys/model/QuestionType.java
#	src/main/java/com/xma/surveys/model/Survey.java
#	src/main/java/com/xma/surveys/services/SurveyManager.java
#	src/test/java/com/xma/surveys/SurveyManagerTest.java
#	src/test/java/com/xma/surveys/model/AnswerTest.java
#	src/test/java/com/xma/surveys/model/QuestionTest.java
#	src/test/java/com/xma/surveys/model/SurveyTest.java
#	src/test/java/com/xma/surveys/model/generators/AnswerGenerator.java
#	src/test/java/com/xma/surveys/model/generators/Generator.java
#	src/test/java/com/xma/surveys/model/statistic/QuestionStatisticTest.java

xma4602 committed:
Merge branch 'data_access' into model

# Conflicts:
#	.gitignore
#	src/main/java/com/xma/surveys/model/Survey.java
#	src/test/java/com/xma/surveys/SurveyManagerTest.java
#	src/test/java/com/xma/surveys/model/generators/SurveyGenerator.java

xma4602 committed:
Переделана модель. Удалены ненужные классы и все, что не относится к модели

xma4602 committed:
Переделаны сервисы

xma4602 committed:
Динамическая подгрузка данных в окно surveys.html

xma4602 committed:
Динамическая подгрузка данных в окно для surveys, questions, answers

xma4602 committed:
Реструктуризация и динамическая загрузка questionnaire.

xma4602 committed:
Добавлены действия на кнопки создания, редактирования и удаления опросов

xma4602 committed:
Добавлены действия на кнопки создания, редактирования и удаления вопросов и ответов

xma4602 committed:
Добавлен Spring

xma4602 committed:
Merge branch 'web_app' into web_access

# Conflicts:
#	src/main/java/com/xma/surveys/entities/QuestionStatus.java
#	src/main/java/com/xma/surveys/entities/QuestionType.java
#	src/test/java/com/xma/surveys/SurveyServiceTest.java
#	src/test/java/com/xma/surveys/generators/AnswerGenerator.java
#	src/test/java/com/xma/surveys/generators/Generator.java

xma4602 committed:
Реструктуризация. Добавлен Конфиг, перенаправляющий запросы

xma4602 committed:
Отлаженные ссылки между страницами

xma4602 committed:
Работающее API на создание опросов

xma4602 committed:
Готовое API для опросов

xma4602 committed:
Доработки API для опросов

xma4602 committed:
рабочие страницы сущностей

xma4602 committed:
Переделано на Spring Data JPA

xma4602 committed:
Работающие страницы для Анкет

xma4602 committed:
Добавлены счетчики в таблицы.

xma4602 committed:
findAll в репозиториях выдает сущности по порядку

xma4602 committed:
Добавлено отображение статистики

KorobetskayaAA reviewed (commented):
В целом смотрится хорошо - и визуально, и в плане организации кода.

Одно большое замечание по контроллерам: у вас перемешаны REST-контроллеры и обычные HTML-контроллеры. Обычно их все-таки разделяют, и все REST-запросы отправляют по какому-нибудь специальному пути, чаще всего /api. Так и код более читабильный, и тестировать, и документировать проще, и пользователю случайно JSON вместо странички не вылетит. Выделите @RestController отдельно, можно даже в отдельную папку их сложить.

Делать вложенные в контроллеры классы для Dto - мне в принципе нравится, но могут быть проблемы: 1) DTO-класс может оказаться переиспользуемым для разных контроллеров; 2) Растет файл контроллера, особенно когда DTO-классов много; 3) Для сложных структур и DTO-класс может быть сложным, и это усугубляет вторую проблему. Здесь оставьте вложенные, но на работе, если в проекте так не делают, не делайте тоже.

Также было бы хорошо файлы bootstrap, раз уж включили в проект, положить в отдельную папку, чтобы не смешивались с вашими.

xma4602 committed:
Контроллеры разделены на View & Rest.
Dto вынесены в отдельную папку.
Обработка ошибок вынесена в ExceptionController
Файлы bootstrap перенесены в подпапку bootstrap

xma4602 commented:
Внес указанные правки

KorobetskayaAA reviewed (approved):
Принято.
